pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java

```diff
@@ -100,7 +100,7 @@ public void write(Record<byte[]> record) {
     }
 
     public KeyValue<String, byte[]> extractKeyValue(Record<byte[]> record) {
-        String key = record.getKey().orElseGet(null);
+        String key = record.getKey().orElseGet(() -> new String(record.getValue()));
```

Comment:
Wouldn't an empty string be more appropriate in this case?

Reply:
Changed.